libraries/standard/http/include/http_client.h

```diff
@@ -1,347 +1,371 @@
+         * parse through of the response as is it receive from the network.
+         * Set to NULL to disable.
+         */
+        HTTPClient_HeaderParsingCallback_t * pHeaderParsingCallback;
+
+        /**
+         * @brief The starting location of the response headers in pBuffer.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        uint8_t * pHeaders;
+
+        /**
+         * @brief Byte length of the response headers in pBuffer.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        size_t headersLen;
+
+        /**
+         * @brief The starting location of the response body in pBuffer.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        uint8_t * pBody;
+
+        /**
+         * @brief Byte length of the body in pBuffer.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        size_t bodyLen;
+
+        /* Useful HTTP header values found. */
+
+        /**
+         * @brief The HTTP response Status-Code.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        uint16_t statusCode;
+
+        /**
+         * @brief The value in the "Content-Length" header is returned here.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        size_t contentLength;
+
+        /**
+         * @brief Count of the headers sent by the server.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        size_t headerCount;
+
+        /**
+         * @brief Flags of useful headers found in the response.
+         *
+         * This is updated by #HTTPClient_Send.
+         */
+        uint32_t flags;
+    } HTTPResponse_t;
 
 /**
  * @brief Initialize the request headers, stored in
- * #HTTPRequestHeaders_t.pBuffer, with initial configurations from 
+ * #HTTPRequestHeaders_t.pBuffer, with initial configurations from
  * #HTTPRequestInfo_t.
- * 
+ *
  * Upon return, #HTTPRequestHeaders_t.headersLen will be updated with the number
- * of bytes written.
- * 
- * TODO: Expand documentation.
- * 
+ * of bytes written if there is sufficient memory.
+ *
+ * Each line in the header is listed below and written in this order:
+ *     <#HTTPRequestInfo_t.method> <#HTTPRequestInfo_t.pPath> <HTTP_PROTOCOL_VERSION>
+ *     User-Agent: <HTTP_USER_AGENT_VALUE>
+ *     Host: <#HTTPRequestInfo_t.pHost>
+ *     Connection: close
+ *
+ * Note that Connection header value can be changed to keep-alive by setting
+ * the HTTP_REQUEST_KEEP_ALIVE_FLAG in #HTTPRequestInfo_t.flags.
+ *
  * @param[in] pRequestHeaders Request header buffer information.
  * @param[in] pRequestInfo Initial request header configurations.
  * @return #HTTP_SUCCESS if successful, an error code otherwise.
- * TODO: Update for exact error codes returned.
+ * @return #HTTP_INVALID_PARAMETER if any provided parameters and their members are NULL.
```

Comment:
All return text should be within the same `@return` tag. Refer here for an example: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/aws/jobs/include/aws_iot_jobs.h#L146-L150